velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -198,5 +202,28 @@ TEST_F(DateTimeFunctionsTest, lastDay) {
   EXPECT_EQ(lastDayFunc(std::nullopt), std::nullopt);
 }
 
+TEST_F(DateTimeFunctionsTest, dateSub) {
+  const auto dateSub = [&](std::optional<int32_t> date,
+                           std::optional<int32_t> value) {
+    return evaluateOnce<int32_t, int32_t>(
+        "date_sub(c0, c1)", {date, value}, {DATE(), INTEGER()});
+  };
+
+  // Check simple tests.
+  int32_t inputDate = parseDate("2019-03-01");
+  int32_t expectedDate = parseDate("2019-02-28");
+  EXPECT_EQ(expectedDate, dateSub(inputDate, 1));
```

Comment:
These tests might be easier to read as

```
EXPECT_EQ('2019-02-28', dateSub('2019-03-01', 1));
```

Reply:
Due to input parameters type limitation, didn't get exactly the same as  EXPECT_EQ('2019-02-28', dateSub('2019-03-01', 1)). Reworked as EXPECT_EQ(parseDate("2019-02-28"), dateSub("2019-03-01", 1)). Kindly help to review again. Thanks.